bin/cml/runner.js

```diff
@@ -229,59 +253,56 @@ const runLocal = async (opts) => {
     dockerVolumes
   });
 
-  const dataHandler = async (data) => {
-    const log = await cml.parseRunnerLog({ data });
-    log && winston.info('runner status', log);
+  proc.stderr.on('data', dataHandler);
+  proc.stdout.on('data', dataHandler);
+  proc.on('disconnect', () =>
+    shutdown({ ...opts, reason: `runner disconnected` })
+  );
+  proc.on('close', (exit) =>
+    shutdown({ ...opts, reason: `runner closed with exit code ${exit}` })
+  );
 
-    if (log && log.status === 'job_started') {
-      RUNNER_JOBS_RUNNING.push({ id: log.job, date: log.date });
-      RUNNER_TIMEOUT_TIMER = 0;
-    } else if (log && log.status === 'job_ended') {
-      const { job } = log;
-
-      const waitCompletedPipelineJobs = () => {
-        return new Promise((resolve, reject) => {
-          try {
-            if (RUNNER_JOBS_RUNNING.length === 1) {
-              resolve([RUNNER_JOBS_RUNNING[0].id]);
-              return;
-            }
-
-            const watcher = setInterval(async () => {
-              const jobs = (
-                await cml.pipelineJobs({ jobs: RUNNER_JOBS_RUNNING })
-              )
-                .filter((job) => job.status === 'completed')
-                .map((job) => job.id);
-
-              if (jobs.length) {
-                resolve(jobs);
-                clearInterval(watcher);
-              }
-            }, 5 * 1000);
-          } catch (err) {
-            reject(err);
+  RUNNER = proc;
+  ({ id: RUNNER_ID } = await cml.runnerByName({ name }));
+
+  if (parseInt(idleTimeout) > 0) {
+    const watcher = setInterval(async () => {
+      let idle = !RUNNER_JOBS_RUNNING.length;
+
+      try {
+        if (cml.driver === 'github') {
+          const job = await cml.runnerJob({ runnerId: RUNNER_ID });
+
+          if (!job && !idle) {
+            winston.error(
+              `Runner should be idle. Reseting jobs. Will try again in ${idleTimeout} secs`
+            );
+
+            RUNNER_JOBS_RUNNING = [];
           }
-        });
-      };
 
-      if (!RUNNER_SHUTTING_DOWN) {
-        const jobs = job ? [job] : await waitCompletedPipelineJobs();
+          if (job && idle) {
+            winston.error(
+              `Runner seems to be busy yet. Will try again in ${idleTimeout} secs`
+            );
 
-        RUNNER_JOBS_RUNNING = RUNNER_JOBS_RUNNING.filter(
-          (job) => !jobs.includes(job.id)
+            idle = false;
+          }
+        }
+      } catch (err) {
+        winston.error(
+          `Error connecting the SCM. Will try again in ${idleTimeout} secs`
         );
+
```

Comment:
```suggestion
        err.message = 'Failed connecting with the SCM; will try again in ${idleTimeout} seconds: ' + err.message;
        winston.error(err);
```

Same as above. Also, why use `--idle-timeout` value for retries? 🤔

Reply:
we have simplified the timer logic. We use the idle-timeout to set the cleanup timer

Reply:
[L304](https://github.com/iterative/cml/pull/689/files#diff-10fb1f5fadda740e153419ffc626eda08260bf2f3b7ba973facefc001d9dcec9R304)? Sounds like an overload of `--idle-timeout` but makes sense.

Reply:
> [L304](https://github.com/iterative/cml/pull/689/files#diff-10fb1f5fadda740e153419ffc626eda08260bf2f3b7ba973facefc001d9dcec9R304)? Sounds like an overload of `--idle-timeout` but makes sense.

Yes. We just only need to run the cleanup at `--idle-timeout` seconds. What we are saying in the logs is that we will retry in the next cleanup.